lib/libzfs_core/libzfs_core.c

```diff
@@ -715,6 +720,20 @@ recv_impl(const char *snapname, nvlist_t *recvdprops, nvlist_t *localprops,
 		if (localprops != NULL)
 			fnvlist_add_nvlist(innvl, "localprops", localprops);
 
+		if (wkeydata != NULL) {
+			/*
+			 * wkeydata must be placed in the special
+			 * ZPOOL_HIDDEN_ARGS nvlist so that it
+			 * will not be printed to the zpool history.
+			 */
```

Comment:
Let's be absolutely sure about this.  Can you add a check to the test case which verifies that this does not appear in the history.